charts/datadog/templates/_container-agent.yaml

```diff
@@ -45,7 +45,7 @@
       value: {{ tpl (.Values.datadog.dogstatsd.tags | join " " | quote) . }}
     {{- end }}
     {{- if not .Values.clusterAgent.enabled }}
-    {{- if .Values.datadog.leaderElection }}
+    {{- if and (not .Values.clusterAgent.enabled) .Values.datadog.leaderElection }}
```

Comment:
is this change needed? The line directly above already ensures the `not .Values.clusterAgent.enabled` condition

Reply:
Indeed, good catch, I must have missed it! Fixed